test/tbb/test_task.cpp

```diff
@@ -840,3 +841,28 @@ TEST_CASE("Check correct arena destruction with enqueue") {
         tbb::finalize(handle, std::nothrow_t{});
     }
 }
+
+//! \brief \ref regression
+TEST_CASE("Check that memory does not leak with static_partitioner + global_control") {
+    tbb::global_control gbl_ctrl{ tbb::global_control::max_allowed_parallelism, std::size_t(tbb::this_task_arena::max_concurrency() / 2) };
+
+    size_t current_memory_usage = 0, previous_memory_usage = 0, stability_counter = 0;
+    bool no_memory_leak = false;
+    std::size_t num_iterations = 100;
+    for (std::size_t i = 0; i < num_iterations; ++i) {
+        for (std::size_t j = 0; j < 100; ++j) {
+            tbb::parallel_for(0, 1000, [] (int) {}, tbb::static_partitioner{});
+        }
+
+        current_memory_usage = utils::GetMemoryUsage();
+        stability_counter = current_memory_usage==previous_memory_usage ? stability_counter + 1 : 0;
+        // If the amount of used memory has not changed during 5% of executions,
+        // then we can assume that the check was successful
+        if (stability_counter > num_iterations / 20) {
+            no_memory_leak = true;
+            break;
+        }
+        previous_memory_usage = current_memory_usage;
+    }
+    REQUIRE_MESSAGE(no_memory_leak, "Seems we get memory leak here.");
```

Comment:
Wrap it into the loop with gradual decrease of the `global_control`'s limit? E.g.,
```cpp
std::size_t current_limit = std::size_t(tbb::this_task_arena::max_concurrency());
while (current_limit /= 2) {
    tbb::global_control gc{ tbb::global_control::max_allowed_parallelism, current_limit };

    // iterations loop goes here {
        // repetitions loop goes here {
        // }
    // }
}
```